### PR TITLE
Added endpoint

### PR DESCRIPTION
**Pull request by FrostShadow6**

### CODE REVIEW

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+require_once '../../../utilities/RSMitemsManagement.php';
+require_once '../../api_headers.php';
+
+// Definitions
+$requestBody = getRequestBody();
+verifyBodyContent($requestBody);
+
+$clientID = getClientID();
+$RStoken =  getRStoken();
+$RSuserID =  getRSuserID();
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:29]:
```suggestion
$RStoken = getRStoken();
$RSuserID = getRSuserID();
```

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+$itemType = $requestBody->itemType;
+$filterProperty   = $requestBody->filterProperty;
+$filterPropertyID = $requestBody->filterPropertyID;
+
+//translateIDs
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:35]:
```suggestion
// translateIDs
```

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+$properties   = array();
+$attributes   = array();
+$results      = array();
+
+// get type of the filterProperty
+$propertyType = getPropertyType($filterProperty, $clientID);
+
+// if filterProperty is unsupported type return empty response
+if (!isSingleIdentifier($propertyType) && !isMultiIdentifier($propertyType)) {
+    returnJsonMessage(200, '');
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:50]: Add a debug mode clause to get more details in the log or in the answer, if debug mode is enabled

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+// Get itemType of the filter property
+$filterItemType = getItemTypeIDFromProperties(array($filterProperty), $clientID);
+
+// verify if item exists
+if (!verifyItemExists($filterPropertyID, $filterItemType, $clientID)) {
+    if ($RSallowDebug) {
+        returnJsonMessage(404, "Item doesn't exist");
+    } else {
+        returnJsonMessage(404, '');
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:61]: This should return 200

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+// Get the value of the property $filterPropertyID for the given $filterProperty
+$valuePropertyRelated = getItemPropertyValue($filterPropertyID, $filterProperty, $clientID);
+
+// get the properties of the itemType
+$properties = getClientItemTypeProperties($itemType, $clientID);
+
+// verify how many items are
+if (strpos($valuePropertyRelated, ',') === false) {
+    //single item identified, return it as usual for backwards compatibility
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:73]:
```suggestion
    // single item identified, return it as usual for backwards compatibility
```

**Review by ifernandez — CHANGES_REQUESTED** [review]

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            $results[] = array(
+                'propertyID'    => $property['id'],
+                'name'  => html_entity_decode($property['name'], ENT_COMPAT, 'UTF-8'),
+                'value' => $value,
+                'type'  => $property['type'],
+                'trs'   => rtrim($trsProperties, ',')
+            );
+        } else {
+            $results[] = array(
+                'propertyID'    => $property['id'],
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:116]:
```suggestion
                'propertyID' => $property['id'],
```

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+                'type'  => $property['type']
+            );
+        }
+    }
+} else {
+    //multiple items, use getFilteredItemsIDs to return all
+
+    // Check if user has permissions to read properties of the item and remove otherwise
+    foreach ($properties as $key => $property) {
+        // fix the id vs ID key issue TODO: review all code and solve it
```
Comment (ifernandez) [inline Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:128]: What does this mean? Is there some pending work in the file?

**Review by ifernandez — CHANGES_REQUESTED** [review]

**Review by ser6l0 — CHANGES_REQUESTED** [review]

File: Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    returnJsonMessage(200, '');
+}
+
+// Get itemType of the filter property
+$filterItemType = getItemTypeIDFromProperties(array($filterProperty), $clientID);
+
+// verify if item exists
+if (!verifyItemExists($filterPropertyID, $filterItemType, $clientID)) {
+    if ($RSallowDebug) {
+        returnJsonMessage(404, "Item doesn't exist");
```
[the remark this answers is not in the document]
Comment (ser6l0) [reply Server/htdocs/AppController/commands_RSM/api/v2/items/getItemFromProperty.php:59]: returnJsonMessage(404, "Source item with ID " . $filterPropertyID . " does not exist");

**Review by ser6l0 — CHANGES_REQUESTED** [review]
Additional comment for line 59

**Review by ser6l0 — APPROVED** [review]

**Pull request merged**